examples/tpm2-seal-unseal/main.go

```diff
@@ -0,0 +1,281 @@
+	}
+	defer func() {
+		if err := rwc.Close(); err != nil {
+			errs = append(errs, fmt.Errorf("can't close TPM %q: %v", tpmPath, err))
+		}
+	}()
+
+	// Create the parent key against which to seal the data
+	srkPassword := ""
+	srkHandle, _, err := tpm2.CreatePrimary(rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", srkPassword, srkTemplate)
+	if err != nil {
+		return append(errs, fmt.Errorf("can't create primary key: %v", err))
+	}
+	defer func() {
+		if err := tpm2.FlushContext(rwc, srkHandle); err != nil {
+			errs = append(errs, fmt.Errorf("unable to flush SRK handle %q: %v", srkHandle, err))
+		}
+	}()
+	fmt.Printf("Created parent key with handle: 0x%x\n", srkHandle)
+
+	// Note the value of the pcr against which we will seal the data
+	pcrVal, err := readPCR(rwc, pcr)
+	if err != nil {
+		return append(errs, fmt.Errorf("unable to read PCR: %v", err))
+	}
+	fmt.Printf("PCR %v value: 0x%x\n", pcr, pcrVal)
+
+	// Get the authorization policy that will protect the data to be sealed
+	objectPassword := "objectPassword"
+	sessHandle, policy, err := policyPCRPasswordSession(rwc, pcr, objectPassword)
+	if sessHandle != tpm2.HandleNull {
+		if flushErr := tpm2.FlushContext(rwc, sessHandle); flushErr != nil {
+			errs = append(errs, fmt.Errorf("unable to flush session: %v", flushErr))
+		} else {
+			sessHandle = tpm2.HandleNull
+		}
+	}
+	if err != nil {
+		errs = append(errs, fmt.Errorf("unable to get policy: %v", err))
+	}
+	if len(errs) > 0 {
+		return
+	}
+	fmt.Printf("Created authorization policy: 0x%x\n", policy)
+
+	// Seal the data to the parent key and the policy
+	dataToSeal := []byte("secret")
+	fmt.Printf("Data to be sealed: 0x%x\n", dataToSeal)
+	privateArea, publicArea, err := tpm2.Seal(rwc, srkHandle, srkPassword, objectPassword, policy, dataToSeal)
+	if err != nil {
+		return append(errs, fmt.Errorf("unable to seal data: %v", err))
+	}
+	fmt.Printf("Sealed data: 0x%x\n", privateArea)
+
+	// Load the sealed data into the TPM.
+	objectHandle, _, err := tpm2.Load(rwc, srkHandle, srkPassword, publicArea, privateArea)
+	if err != nil {
+		return append(errs, fmt.Errorf("unable to load data: %v", err))
+	}
+	defer func() {
+		if err := tpm2.FlushContext(rwc, objectHandle); err != nil {
+			errs = append(errs, fmt.Errorf("unable to flush object handle %q: %v", objectHandle, err))
+		}
+	}()
+	fmt.Printf("Loaded sealed data with handle: 0x%x\n", objectHandle)
+
+	// Unseal the data
+	unsealedData, unsealErr, otherErrs := unseal(rwc, pcr, objectPassword, objectHandle)
+	if unsealErr != nil {
+		errs = append(errs, fmt.Errorf("unable to unseal data: %v", unsealErr))
+	}
+	for otherErr := range otherErrs {
+		errs = append(errs, fmt.Errorf("error while unsealing data: %v", otherErr))
+	}
+	if len(errs) > 0 {
+		return
+	}
+	fmt.Printf("Unsealed data: 0x%x\n", unsealedData)
+
+	// Try to unseal the data with the wrong password
+	_, unsealErr, otherErrs = unseal(rwc, pcr, "wrong-password", objectHandle)
```

Comment:
This seems out of place.
This file should be an example of correct usage, not an integration test.